app/backend/src/couchers/servicers/groups.py

```diff
@@ -199,10 +211,14 @@ def JoinGroup(self, request, context):
             if not cluster:
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.GROUP_NOT_FOUND)
 
-            current_membership = cluster.members.filter(User.id == context.user_id).one_or_none()
-            if current_membership:
+            user_in_group = cluster.members.filter(User.id == context.user_id).one_or_none()
+            if user_in_group:
                 context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.ALREADY_IN_GROUP)
 
+            user_visible = session.query(User).filter(User.is_visible).filter(User.id == context.user_id).one_or_none()
+            if not user_visible:
+                context.abort(grpc.StatusCode.NOT_FOUND, errors.USER_NOT_FOUND)
+
```

Comment:
No need to check if `context.user_id` exists/is visible/etc. That needs to be handled in auth